### ISSUE TITLE
Sort by Rarity does not re-sort

### ISSUE THREAD

**Issue by MBloom94** [DestinyItemManager/DIM#361]
"Sort items by rarity" currently broken and does not ever change the sorting of items.

v3.2.0 Beta on Chrome v47.0.2526.106 m (64-bit)

**Comment by SunburnedGoose** [comment]
Can confirm.  Changing sort does not affect the UI.

**Comment by bhollis** [comment]
I'll take a look. I specifically tested this.

**Comment by bhollis** [comment]
OK, turns out this was  just a typo on my part. But while I was digging into the problem I found some things that could be tweaked, and now the performance of changing settings is greatly improved. Also, to apologize for the bug, I changed "rarity" sorting to sort items of the same rarity by primary stat. Cheers!

**Comment by Mulchman** [comment]
Why primary stat and not alphabetical? I personally dislike seeing (eg.)
Zhalo before Bad Juju...

**Comment by SunburnedGoose** [comment]
:)

**Comment by bhollis** [comment]
Because, in general, the weapon you want to equip is the one with the
highest attack.

**Comment by Mulchman** [comment]
This is a bad assumption to make. Sure, it's safe to say users put on their
highest light when decrypting (at least to get to a minimum of 311 or their
highest light if they're still leveling up), but in general players want to
quickly look at the listing of items to grab their favorite weapon for the
activity they want to do.

For example, when I want to do PvP, I look at exotic primaries and grab
TLW/Thorn/MIDA and I know where they're going to be because I'm used to
exotics being grouped together and the fact that alphabetical sorting is
intuitive and something we've been doing our entires lives. Sorting by
primary stat makes no sense - MIDA/TLW are going to be somewhere near the
top and way far away from Thorn - and it now turns things in to a scavenger
hunt. It also makes the scavenger hunt dynamic as I bring both of those
weapons closer to 320; I'm pretty confident that no one likes a non stable
sort.

In conclusion: it's not intuitive and it's not stable.

**Comment by SunburnedGoose** [comment]
Let's split this off into a new issue and for both rarity and stats, we can alpha sort the list initially and then do the subsequent rarity/stat sort.

**Comment by bhollis** [comment]
Feel free to revert 146142a5458f5c4e939eba36e065d1ab4eedda93 if it's gonna cause consternation. I certainly wouldn't use sort by rarity without it.

**Comment by bhollis** [comment]
Also, yes, the sort is stable because we still sort by name under the sort by primary.
